File: codegen/results/v001/qwen3.5_27b_nothink/20260408_103648/typescript/003_fibonacci/solution.ts
function main(): void {
  // Parse the integer argument from the command line
  const input = process.argv[2];
  
  if (!input) {
    return;
  }

  const N = parseInt(input, 10);

  // If N < 1, print nothing
  if (isNaN(N) || N < 1) {
    return;
  }

  // Start the sequence: 1, 1, 2, 3, 5, ...
  let a = 1;
  let b = 1;

  // Print the first Fibonacci number if it doesn't exceed N
  if (a <= N) {
    console.log(a);
  }

  // If N is 1, the loop below won't print anything else (b=1, a becomes 2)
  // But we need to handle the sequence correctly.
  // The sequence is 1, 1, 2... so we print the first 1.
  // Then we calculate next numbers.
  
  while (b <= N) {
    console.log(b);
    const next = a + b;
    a = b;
    b = next;
  }
}

main();

File: codegen/results/v001/qwen3.5_27b_nothink/20260408_103648/go/003_fibonacci/solution.go
package main

import (
	"fmt"
	"os"
	"strconv"
)

func main() {
	if len(os.Args) != 2 {
		return
	}

	n, err := strconv.Atoi(os.Args[1])
	if err != nil {
		return
	}

	if n < 1 {
		return
	}

	a, b := 1, 1

	// Print the first number if it's within range
	if a <= n {
		fmt.Println(a)
	}

	// If N is 1, we are done because the second number is also 1
	// but the sequence is 1, 1. The loop below handles the second 1
	// because b starts at 1.
	
	for b <= n {
		fmt.Println(b)
		a, b = b, a+b
	}
}